packages/foam-core/src/markdown-provider.ts

```diff
@@ -129,20 +134,25 @@ export function stringifyMarkdownLinkReferenceDefinition(
 }
 export function createMarkdownReferences(
   graph: NoteGraph,
-  noteId: ID,
+  inputNote: GraphNote,
```

Comment:
was wondering, why are you passing the `Note` object instead of its `ID`?

Reply:
It seems like `createIdFromURI` in class `NoteGraph` is a private funtion, meaning the ability of turning `URI` to `ID` is restricted to `NoteGraph`. Instead of dealing with (private) hash value, I think it would be better to just pass the entire node object. An alternate solution would be to make the translation function public.

Reply:
yes you are right on both fronts regarding `createIdFromURI`, I think adding the function you did is a good approach.
What I was referring to is that I saw you create a `Note` in `generateReferenceList` (`wikilink-reference-generation.ts:138`). 
I think that you could pass `note.id` when calling this function a few lines later (line 141), but I saw you pass the whole `Note` object, so was wondering why.
I don't have a strong preference, was mostly curious to understand :)

Reply:
Woops... I did not think about that... Thank you for point it out :) I will revert back to the original solution in the new PR.